libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -244,6 +283,19 @@ void ContractCompiler::appendConstructor(FunctionDefinition const& _constructor)
 	_constructor.accept(*this);
 }
 
+void ContractCompiler::appendCallPreventer(ContractDefinition const& _contract)
+{
+	solAssert(_contract.isLibrary(), "");
+	// Special constant that will be replaced by the address at deploy time.
+	// This has to be as complicated as possible and only used once
+	// so that the optimizer will not change it.
```

Comment:
That sounds more than a bit brittle...

Reply:
Note that there is another runtime check for the first opcode to be a `PUSH20` opcode, but it is not ideal, no.